src/jaxsim/api/com.py

```diff
@@ -0,0 +1,233 @@
+import jax
+import jax.numpy as jnp
+import jaxlie
+
+import jaxsim.api as js
+import jaxsim.math
+import jaxsim.typing as jtp
+
+from .common import VelRepr
+
+
+@jax.jit
+def com_position(
+    model: js.model.JaxSimModel, data: js.data.JaxSimModelData
+) -> jtp.Vector:
+    """
+    Compute the position of the center of mass of the model.
+
+    Args:
+        model: The model to consider.
+        data: The data of the considered model.
+
+    Returns:
+        The position of the center of mass of the model w.r.t. the world frame.
+    """
+
+    m = js.model.total_mass(model=model)
+
+    W_H_L = js.model.forward_kinematics(model=model, data=data)
+    W_H_B = data.base_transform()
+    B_H_W = jaxlie.SE3.from_matrix(W_H_B).inverse().as_matrix()
+
+    def B_p̃_LCoM(i) -> jtp.Vector:
+        m = js.link.mass(model=model, link_index=i)
+        L_p_LCoM = js.link.com_position(
+            model=model, data=data, link_index=i, in_link_frame=True
+        )
+        return m * B_H_W @ W_H_L[i] @ jnp.hstack([L_p_LCoM, 1])
+
+    com_links = jax.vmap(B_p̃_LCoM)(jnp.arange(model.number_of_links()))
+
+    B_p̃_CoM = (1 / m) * com_links.sum(axis=0)
+    B_p̃_CoM = B_p̃_CoM.at[3].set(1)
+
+    return (W_H_B @ B_p̃_CoM)[0:3].astype(float)
+
+
+@jax.jit
+def com_linear_velocity(
+    model: js.model.JaxSimModel, data: js.data.JaxSimModelData
+) -> jtp.Vector:
+    """
+    Compute the linear velocity of the center of mass of the model.
+
+    Args:
+        model: The model to consider.
+        data: The data of the considered model.
+
+    Returns:
+        The linear velocity of the center of mass of the model w.r.t. the world frame.
+
+    Note:
+        Given that the center of mass is a point, we cannot define a 6D velocity
+        corresponding to any specific frame. This method returns a 3D velocity that
+        would correspond to the linear part of the 6D velocity of a generic frame
+        having the center of mass as origin.
+    """
+
+    with data.switch_velocity_representation(VelRepr.Mixed):
+        W_ṗ_G = average_centroidal_velocity(model=model, data=data)[0:3]
+        return W_ṗ_G
```

Comment:
What happens for others velocity representation?

Reply:
This method always returns ${}^W \dot{\mathbf{p}}_G$, where $G = ({}^W \mathbf{p}\_{\text{CoM}}, [W])$ like iDynTree.

Reply:
Do you think that it can be more uniform returning instead ${}^{G[W]}\boldsymbol{v}\_{W,G}$ in inertial-fixed and mixed representations, and ${}^{G[B]}\boldsymbol{v}\_{W,G}$ in body-fixed? There are some methods in iDynTree (and also here since I maintained the approach) doing that.

Reply:
Yes, that could make sense. Anyhow, I was actually confused by the `data.switch_velocity_representation`, but after checking it out now everything is clear.

Reply:
@traversaro I'm going to merge this PR with `jaxsim.api.com.com_linear_velocity` always returning ${}^W \dot{\mathbf{p}}_{\text{CoM}}$. We can discuss if it's sound returning something else in body-fixed coordinates in #108.

Reply:
Ok!

Reply:
By the way, I looked in iDynTree, and to me it seems that always returning ${}^W \dot{p}_{CoM}$ https://github.com/robotology/idyntree/blob/7e029b963d44fa2606bd608019ba9b82a4398741/src/high-level/src/KinDynComputations.cpp#L1941-L1958 is a bug/inconsistency in iDynTree, that we may want to fix.

Reply:
Ok if this is the intended usage, then I'll update directly this PR instead of doing it in #108.

Reply:
I forced pushed this fix, @traversaro can you please confirm that https://github.com/ami-iit/jaxsim/compare/3b63bb8cf6962c6e1e0f71e5120da480964d3f51..b3864af61e37c48ca24a83146a0f96b0d3063dab is what you had in mind?